charts-community-modules/ag-charts-community/src/chart/chart.ts

```
@@ -733,32 +764,32 @@ export abstract class Chart extends Observable implements AgChartInstance {
         return new Map(labels.map((l, i) => [visibleSeries[i], l]));
     }
 
+    private attachLegend(legendType: string) {
+        if (this.legendType === legendType || !(legendType in this.legendFactories)) {
+            return;
+        }
+        const legendFactory = this.legendFactories[legendType];
+        this.legend?.destroy();
+        const ctx = this.getModuleContext();
+        this.legend = legendFactory(ctx);
+        this.legend.attachLegend(this.scene.root);
+    }
+
     private async updateLegend() {
         const legendData: LegendDatum[] = [];
-
+        let legendType = 'category';
         this.series
             .filter((s) => s.showInLegend)
-            .forEach((series) => {
-                legendData.push(...series.getLegendData());
+            .forEach((series, i) => {
+                const data = series.getLegendData();
+                if (data && data.length > 0 && (i === 0 || data[0].legendType === legendType)) {
+                    legendData.push(...data);
+                    legendType = data[0].legendType;
+                }
             });
```

I wonder if it would be better to just call all `series.getLegendData()` and then inspect the results, rather than trying to keep track of the first legendType as we go?

It seems like if there are mismatching `legendType`s that we should throw an error perhaps? And if that check passes, we can simply concatenate all of the `data` items together?